src/nvim/eval.c

```diff
@@ -12231,19 +12231,23 @@ static void get_user_input(typval_T *argvars, typval_T *rettv, int inputdialog)
   if (prompt != NULL) {
     /* Only the part of the message after the last NL is considered as
      * prompt for the command line */
-    p = vim_strrchr(prompt, '\n');
-    if (p == NULL)
+    if (cmdline_get_external()) {
```

Comment:
Maybe this logic should be moved to `getcmdline_prompt`. It would seem cleaner to have all display logic in `ex_getln.c` and not in `eval.c` (and remove the need for `get_cmdline_external()` )